profile/src/test/java/com/scottlogic/deg/profile/reader/file/names/NameRetrievalServiceTest.java

```diff
@@ -52,13 +47,31 @@ public void retrieveValuesFirst() {
         assertEquals(names, namesToHolders("Mark", "Paul", "Jolene", "Tanya"));
     }
 
+    private void mockFirst() {
+        mockNames("names/firstname_male.csv", "Mark", "Paul");
+        mockNames("names/firstname_female.csv", "Jolene", "Tanya");
+    }
+
     @Test
     public void retrieveValuesLast() {
         mockLast();
         Set<NameHolder> names = service.retrieveValues(NameConstraintTypes.LAST);
         assertEquals(names, namesToHolders("Gore", "May"));
     }
 
+    private void mockLast() {
```

Comment:
i don't understand the point of separating the mocks

Reply:
JUnit 5 will throw an error if you have unused mocks in each test case. It enforces that you should only mock what you use in the given test.

Reply:
that sounds pointless, just mock them all

Reply:
> that sounds pointless, just mock them all

I think you've misunderstood. JUnit 5 will throw an exception during a test if by the end of the test runtime there are mocks or overridden stubs that are unused in that test method. Hence, it is not possible to extract out the mocks because at least two tests have no shared mocks.